packages/commonwealth/client/scripts/state/api/user/updateUser.ts

```diff
@@ -0,0 +1,69 @@
+import { DEFAULT_NAME } from '@hicommonwealth/shared';
+import MinimumProfile from '../../../models/MinimumProfile';
+import { trpc } from '../../../utils/trpcClient';
+import useUserStore from '../../ui/user';
+import { ApiEndpoints, queryClient } from '../config';
+
+interface AddressesWithChainsToUpdate {
+  address: string;
+  chain: string;
+}
+
+interface UseUpdateProfileByAddressMutation {
+  addressesWithChainsToUpdate?: AddressesWithChainsToUpdate[];
+}
+
+const useUpdateUserMutation = ({
+  addressesWithChainsToUpdate,
+}: UseUpdateProfileByAddressMutation = {}) => {
+  const user = useUserStore();
+
+  return trpc.user.updateUser.useMutation({
+    onSuccess: (updated) => {
+      addressesWithChainsToUpdate?.map(({ address, chain }) => {
+        const key = [ApiEndpoints.FETCH_PROFILES_BY_ADDRESS, chain, address];
+        const existingProfile = queryClient.getQueryData(key);
+
+        // TEMP: since we don't get the updated data from API, we will cancel existing and refetch profile
+        // data for the current specified adddress/chain key pair
+        if (existingProfile) {
+          const updatedProfile = new MinimumProfile(address, chain);
+          updatedProfile.initialize(
+            updated.id!,
+            updated.profile.name!,
+            address,
+            updated.profile.avatar_url!,
+            chain,
+            null,
+          );
+          queryClient.setQueryData(key, () => updatedProfile);
+        }
+      });
+
+      // if `userId` matches auth user's id, refetch profile-by-id query for auth user.
+      if (user.id === updated.id) {
+        const keys = [
+          [ApiEndpoints.FETCH_PROFILES_BY_ID, undefined],
+          [ApiEndpoints.FETCH_PROFILES_BY_ID, user.id],
+        ];
+        keys.map((key) => {
+          queryClient.cancelQueries(key).catch(console.error);
```

Comment:
yes we should update cache, thanks for catching

Reply:
so, should we change something here? not clear to me how this works. @masvelio @mzparacha

Reply:
oh ok so i just realised that this piece of code was from here https://github.com/hicommonwealth/commonwealth/pull/8574/files/69819d074e5e89ac491f3905631398b21a2d46ac#diff-52b6c1c828b913c9b791820d16005ca3064f854102b97afef883e12f2512b44bL95-L98

and it was intentional to re-query the data as the response format was different at that time. Also its better to refetch data at this point then to update cache

I suggest we don't do anything here

cc: @masvelio @Rotorsoft

Reply:
I think it's probably better if you guys commit the changes directly to this branch to avoid the back and forth ... I'm still trying to understand the client patterns we are using here